### DISCUSSION TITLE
Contribution for Côte d’Ivoire/Ivory Coast

### DISCUSSION THREAD

**Discussion by GilbertKouame** [electricitymaps/electricitymaps-contrib#7124] [General]
Hi guys,

We are trying to contribute to electricity map by uploading the information for Côte d’Ivoire. But having some trouble for step 4 : make code changes and test them.
We don't see where to click to (or how to make) the changes to the code. Any help would be appreciated.

**Comment by VIKTORVAV99** [comment]
Hi!

I'm not exactly sure what you mean?

If you are trying to add data for a new zone someone will have to build a parser for it so we can continuously fetch new data.

But there is no button or similar for this as every source use a different data structure so each parser have to be custom built for that source.

**Comment by GilbertKouame** [comment]
Ok I'll try to make it clear. We are following the 6 steps to make a code
contribution.
It's said :

1/ Fork the repository : done
2/ Clone your fork of the repository : done
3/Set up and start your local environment (see steps here) : done

4/ Make your code changes and test them in your local environment : we
don't see how to do that. And this is we are.

So, We should instead build a parser. Right ?

Thanks for your reply.

Gilbert



Le ven. 30 août 2024, 10:36, Viktor Andersson ***@***.***> a
écrit :

**Reply by VIKTORVAV99** [reply]
I think you need to take a look at https://github.com/electricitymaps/electricitymaps-contrib/wiki/Building-a-new-parser it should describe what is needed to build a new parser.

**Comment by GilbertKouame** [comment]
By the way, another question ...

We are contributing with annual data right hoping it'll incentivize
decision makers here to cooperate for the real time data ... at l'est
monthly or daily.

Many thanks.
Gilbert

Le ven. 30 août 2024, 10:36, Viktor Andersson ***@***.***> a
écrit :

**Reply by VIKTORVAV99** [reply]
We don't accept annual data simply because it's not accurate enough when compared to hourly data and because the delay for getting such data would be too long.

This page lists all the requirements https://github.com/electricitymaps/electricitymaps-contrib/wiki/Technical-requirements-for-parser-data

**Reply by GilbertKouame** [reply]
I am a bit surprised by your answer because electricity map is currently using new models to extrapolate monthly and yearly data to have hourly informations. So how can you say that you don't accept yearly data ?

On another note, the data on Ghana is not correct. The electricity generation mix in Ghana is hydro, solar, gas and fioul. You guys omitted solar.

Your project is a beautiful one so I really want to help but you leave me with the impression that you don’t need the help of local people or country-specific information. If that is the case then OK I have no choice than stop contributing.

Gilbert

**Reply by VIKTORVAV99** [reply]
First that reply was made 4 months ago, a lot has changed since then and the decision to use yearly and monthly data was made after that.

Secondly the estimation system and model don't exist in this repository so you can't contribute directly with building a parser, so the same requirements for parsers stand.

If you have a data source you can share that would be great but we are currently expanding these zones based on internal prioritizations, data availability and output quality.

**Reply by VIKTORVAV99** [reply]
Also there is solar estimated for Ghana:
<img width="588" alt="image" src="https://github.com/user-attachments/assets/d52a4593-aa2e-40a1-be57-80ec6ecd6b27" />

**Comment by GilbertKouame** [comment]
First, public electricity data (yearly) in West Africa are generally two
years late. I mean the most up-to-date data in 2024 will be from 2022. And
those information will be put in the annual report. I joined a report from
Ghana with statistics from 2000 to 2022.

Second, I also joined a file from Ivory Coast with electricity statistics
from 2022.

Third, Ghana and Ivory Coast exchange electricity to each other. So maybe
the flow tracing technique will be required to delineate carbon intensity
of each country. I made the calculation and got 322 and 385 g/kWh for
respectively Ghana and Ivory Coast in 2022.

Lastly I don’t know where you get the image you just already sent but from
my side (image attached) I don’t see the data on solar. That’s why I said
you did not have solar but OK if you have it no problem for me.


Le sam. 4 janv. 2025, 22:10, Viktor Andersson ***@***.***> a
écrit :

**Reply by VIKTORVAV99** [reply]
The image is monthly data from our app so it's visible for all, if there data is delayed there is not much we can do about that and it will simply have to be updated retroactively when it's available.

The current models we have also take weather patterns into account so there will be some flexibility and seasonality even in the estimation models.

Exchange data will be a lot trickier to estimate on such a large timescale and probably not something we will be able to do in the near future, so we'd need realtime data for the exchange to take that into account when it comes to carbon intensity as that is counted on a hourly level and then aggregated.

Do you have a link to these reports?
It would be interesting to compare it to our own data.

**Comment by GilbertKouame** [comment]
Hi Viktor,


Here are the links to the documents you asked for.

From Ghana
https://www.energycom.gov.gh/index.php/media-center/latest-news/239-national-energy-statistical-bulletin-2023

From Ivory Coast

https://data.gouv.ci/datasets/evolution-de-la-production-brute-annuelle-d'electricite-de-source-thermique-and-hydraulique-en-mwh-de-1980-a-2018

https://data.gouv.ci/datasets/echanges-d'electricite-depuis-ou-vers-la-cote-d'ivoire

The statistics in these links above (Ivory Coast) stop in 2018. So the
document I sent to you about Ivory Coast in 2022 is not available online.

The data from 2019 to 2022 are not available online. I just have them
because I have close friends in the national energy office (Direction
Générale de l’Énergie). So every time there is a new document validated and
that can be made publicly available they send it to me.


Gilbert




Le sam. 4 janv. 2025, 22:57, Viktor Andersson ***@***.***> a
écrit :

**Comment by GilbertKouame** [comment]
Hi victor,

Here is an additional information on Ivory Coast.

This report below shows the electricity production by type of fuel used.
This is where you can see the amount of gas and heavy oil used in the
thermal plants.

I am sharing this because everything denominated gas in electricity
production is not only gas, it is also heavy oil used from time to time and
this report tells the amount used.

https://anare.ci/download/anare-ci-rapport-dactivites-2022/


Regards,
Gilbert
